### DISCUSSION TITLE
SQLite Server (libsql) using Lambdas with Lambda-Web-Adapter

### DISCUSSION THREAD

**Discussion by gabriels1234** [awslabs/aws-lambda-web-adapter#517] [General]
Hi @bnusunny, team.

As you know, it's not as straightforward to have lamdas that use SQL databases (especially when budget is tight).

I was wondering if lambda-web-adapter paired with libsql (client/server for sqlite) using an S3 backend (see bottomless) could work.

The implications could be huge. It could be the first serverless serverless-database. libsql (server) works over HTTPS.

What are your thoughts?

Some useful links:

[LibSQL]
[main] github: tursodatabase/libsql
[server] guthub: tursodatabase/libsql/tree/main/libsql-server
[bottomless, S3] github: tursodatabase/libsql/tree/main/bottomless   
[client for django] github: /aaronkazah/django-libsql

**Comment by bnusunny** [comment]
This is a very interesting! But I would not couple this with LWA. It could be a seperate extension all by it own.

**Comment by sladg** [comment]
@gabriels1234 how does bottomless/turso handle concurrent access? I wasn't able to figure out from their doc what happens if multiple lambdas would run at the same time. Sqlite on typical FS would have first process lock it. This would not apply if multiple Sqlite files exists tho (multiple lambdas download file from underlying S3). Could you elaborate?
